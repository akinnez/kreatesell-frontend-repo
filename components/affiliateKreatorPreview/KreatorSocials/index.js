import {
	AiFillFacebook,
	AiFillInstagram,
	AiFillLinkedin,
	AiFillTwitterSquare,
	AiOutlineWhatsApp,
} from 'react-icons/ai';
import styles from './index.module.scss';

const KreatorSocials = ({facebook, instagram, linkedIn, mobile, twitter}) => (
	<div className={styles.kreator__socials}>
		<span>Connect with me on</span>
		<div>
			{facebook && (
				<a href={facebook} target="_blank" rel="noreferrer">
					<AiFillFacebook />
				</a>
			)}
			{instagram && (
				<a href={instagram} target="_blank" rel="noreferrer">
					<AiFillInstagram />
				</a>
			)}
			{linkedIn && (
				<a href={linkedIn} target="_blank" rel="noreferrer">
					<AiFillLinkedin />
				</a>
			)}
			<a
				href={`https://wa.me/${mobile.replace(/[\+\-\(\) ]/g, '')}`}
				target="_blank"
				rel="noreferrer"
			>
				<AiOutlineWhatsApp />
			</a>
			{twitter && (
				<a href={twitter} target="_blank" rel="noreferrer">
					<AiFillTwitterSquare />
				</a>
			)}
		</div>
	</div>
);

export default KreatorSocials;
