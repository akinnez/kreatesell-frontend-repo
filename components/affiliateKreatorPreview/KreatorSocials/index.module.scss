@use "public/css/colors";
@use "public/css/mixins";
@use "public/css/screens";

.kreator__socials {
  @include mixins.flex(row, flex-start, center);
  flex-wrap: wrap;
  gap: 5px 10px;
  flex: 0 0 310px;

  span {
    font-size: 14px;
    color: colors.$base-gray;
  }

  div {
    @include mixins.flex(row, space-between, center);
    column-gap: 15px;
  }

  a {
    color: colors.$black-100;

    > svg {
      font-size: 21px;
    }
  }
}

@include screens.minScreen("medium") {
  .kreator__socials {
    justify-content: center;
  }
}

@include screens.minScreen("tablet") {
  .kreator__socials {
    flex-wrap: nowrap;
  }
}
