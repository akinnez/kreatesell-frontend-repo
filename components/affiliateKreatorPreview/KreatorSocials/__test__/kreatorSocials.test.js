import KreatorSocials from '..';
import renderer from 'react-test-renderer';

const mockProps = {
	facebook: 'mockFacebook-url',
	instagram: 'mockInstagram-url',
	linkedIn: 'mockIinstagram - url',
	mobile: 'mockMobile - url',
	twitter: 'mockTwitter-url',
};

it('renders a snapshot of <KreatorSocials />', () => {
	const socials = renderer.create(<KreatorSocials {...mockProps} />).toJSON();
	expect(socials).toMatchSnapshot();
});
