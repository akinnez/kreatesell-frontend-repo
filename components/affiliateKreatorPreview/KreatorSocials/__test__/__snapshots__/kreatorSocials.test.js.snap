// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <KreatorSocials /> 1`] = `
<div
  className="kreator__socials"
>
  <span>
    Connect with me on
  </span>
  <div>
    <a
      href="mockFacebook-url"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
        />
      </svg>
    </a>
    <a
      href="mockInstagram-url"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 0 1-47.9 47.9z"
        />
      </svg>
    </a>
    <a
      href="mockIinstagram - url"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
        />
      </svg>
    </a>
    <a
      href="https://wa.me/mockMobileurl"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        t="1569683925316"
        version="1.1"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M713.5 599.9c-10.9-5.6-65.2-32.2-75.3-35.8-10.1-3.8-17.5-5.6-24.8 5.6-7.4 11.1-28.4 35.8-35 43.3-6.4 7.4-12.9 8.3-23.8 2.8-64.8-32.4-107.3-57.8-150-131.1-11.3-19.5 11.3-18.1 32.4-60.2 3.6-7.4 1.8-13.7-1-19.3-2.8-5.6-24.8-59.8-34-81.9-8.9-21.5-18.1-18.5-24.8-18.9-6.4-0.4-13.7-0.4-21.1-0.4-7.4 0-19.3 2.8-29.4 13.7-10.1 11.1-38.6 37.8-38.6 92s39.5 106.7 44.9 114.1c5.6 7.4 77.7 118.6 188.4 166.5 70 30.2 97.4 32.8 132.4 27.6 21.3-3.2 65.2-26.6 74.3-52.5 9.1-25.8 9.1-47.9 6.4-52.5-2.7-4.9-10.1-7.7-21-13z"
        />
        <path
          d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3-41.3-41.3-89.5-73.8-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6 0.3-119.3 12.3-174.5 35.9-53.3 22.8-101.1 55.2-142 96.5-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9 0.3 69.4 16.9 138.3 48 199.9v152c0 25.4 20.6 46 46 46h152.1c61.6 31.1 130.5 47.7 199.9 48h2.1c59.9 0 118-11.6 172.7-34.3 53.5-22.3 101.6-54.3 142.8-95.2 41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5 0.3-60.9-11.5-120-34.8-175.6z m-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-0.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-0.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-0.6 99.6-39.7 192.9-110.1 262.7z"
        />
      </svg>
    </a>
    <a
      href="mockTwitter-url"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM727.3 401.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
        />
      </svg>
    </a>
  </div>
</div>
`;
